DoctorWho.Web/Controllers/Services/EpisodeService.cs

```diff
@@ -12,14 +12,32 @@ namespace DoctorWho.Web.Controllers.Services
     public class EpisodeService : IEpisodeService
     {
         private readonly IEpisodeRepository episodeRepository;
+        private readonly IDoctorRepository doctorRepository;
+        private readonly IAuthorRepository authorRepository;
         private readonly IUnitOfWork unitOfWork;
-
-        public EpisodeService(IEpisodeRepository episodeRepository, IUnitOfWork unitOfWork)
+        public EpisodeService(IEpisodeRepository episodeRepository, IDoctorRepository doctorRepository, IUnitOfWork unitOfWork, IAuthorRepository authorRepository)
         {
             this.episodeRepository = episodeRepository;
+            this.doctorRepository = doctorRepository;
+            this.authorRepository = authorRepository;
             this.unitOfWork = unitOfWork;
         }
 
+        public async Task<int> CreateEpisodeAsync(Episode episode)
+        {
+            if(!doctorRepository.DoctorExists(episode.DoctorId))
+            {
+                return -1;
```

Comment:
can't we throw a useful exception so we can provide the end user with a meaning full exception ?